### DISCUSSION TITLE
what are the encryption algorithm used for secint function

### DISCUSSION THREAD

**Discussion by gro41** [lschoe/mpyc#48] [Q&A] [answered]
what are the encryption algorithm used for secint function

**Comment by lschoe** [comment, marked as answer]
The short answer is: Shamir secret sharing.

**Reply by spyder73** [reply]
And what is the degree of the polynomial chosen? 
If, N = amount of parties - is the degree of the polynomial then t = N-1, so that all parties are necessary for the reconstruction of the secret? 
or is it t = N/2, since

The MPC protocols are based on Shamir’s threshold secret sharing scheme and **withstand passive adversaries controlling less than half of the parties.**

Best regards
Spyder

**Comment by lschoe** [comment]
Hi @spyder73. So, for $m$ parties tolerating at most $t$ corrupt parties, $t<m/2$ is required, which can be phrased as "an adversary controlling less than half of the parties." This enables the use of Shamir secret sharing for secure multiplication of two secret-shared values, which involves a product polynomial of double degree $2t$ shared between $m$ parties. Such a product polynomial of degree $2t$ can only be reconstructed from $2t+1$ shares, which is another way to understand why $m\geq2t+1$, which is equivalent to $t<m/2$, is required.

To tolerate $t=1$ corrupt party we thus need $m\geq3$ parties, which means that we have an honest majority.

**Reply by spyder73** [reply]
thank you very much! this means the protocol "only" functions with an honest majority, because if we have less we can not reconstruct the secret?

**Reply by lschoe** [reply]
Right, one wouldn't be able to reconstruct any output after at least one secure multiplication has been performed. If we only do secure additions then $t$ can be anything between 0 and $m-1$ , but that's not very interesting. Also note that we assume passively corrupt parties (a.k.a. semi-honest or honest-but-curious parties); to withstand $t$ actively corrupt parties one generally requires $t<m/3$.
